### PR TITLE
新增API：paddle.nn.functional.sparse_attention

### PR DESCRIPTION
**Pull request by Liu-xiandong**
给paddle添加API：paddle.nn.functional.sparse_attention
中文预览效果：http://10.136.157.23:8090/documentation/docs/zh/api/paddle/nn/functional/sparse_attention_cn.html

### 示例：
```
import paddle
import numpy as np

query_data = np.array([[[[0, 1,], [2, 3], [ 0, 1], [2, 3]]]]).astype("float32")
key_data = np.array([[[[0, 1,], [2, 3], [ 0, 1], [2, 3]]]]).astype("float32")
value_data = np.array([[[[0, 1,], [2, 3], [ 0, 1], [2, 3]]]]).astype("float32")
sparse_csr_offset_data = np.array([[[0, 2, 4, 6, 8]]]).astype("int32")
sparse_csr_columns_data = np.array([[[0, 1, 0, 1, 2, 3, 2, 3]]]).astype("int32")
print(query_data.shape)
# (1, 1, 4, 2)
print(sparse_csr_offset_data.shape)
# (1, 1, 5)
print(sparse_csr_columns_data.shape)
# (1, 1, 8)
paddle.disable_static()
query = paddle.to_tensor(query_data, stop_gradient=False, place=paddle.CUDAPlace(0))
key = paddle.to_tensor(key_data, stop_gradient=False, place=paddle.CUDAPlace(0))
value = paddle.to_tensor(value_data, stop_gradient=False, place=paddle.CUDAPlace(0))
offset = paddle.to_tensor(sparse_csr_offset_data, stop_gradient=False, place=paddle.CUDAPlace(0))
columns = paddle.to_tensor(sparse_csr_columns_data, stop_gradient=False, place=paddle.CUDAPlace(0))
output = paddle.nn.functional.sparse_attention(query, key, value, offset, columns)
print(output)

# [[[[1.60885942, 2.60885954],
#       [1.99830270, 2.99830270],
#       [1.60885942, 2.60885954],
#       [1.99830270, 2.99830270]]]]

### CODE REVIEW

File: docs/api/paddle/nn/functional/sparse_attention_cn.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+该OP对Transformer模块中的Attention矩阵进行了稀疏化，从而减少内存消耗和计算量。
+
+其稀疏数据排布通过CSR格式表示，CSR格式包含两个参数， ``offset`` 和 ``colunms`` 。
+
+.. math::
+   result=softmax(\frac{ Q * K^T }{\sqrt{d}}) * V
+
+其中，``Q``，``K``，``V`` 表示注意力模块的三个输入参数。这三个参数的维度是一样的。 ``d`` 代表这三个参数的最后一个维度的大小。
+
+参数：
```
Comment (dingjiaweiww) [inline docs/api/paddle/nn/functional/sparse_attention_cn.rst:17]: 文档的格式有点问题，建议参考下中文文档的写法，参数和返回都有规定具体格式：参考文档：http://agroup.baidu.com/paddlepaddle/md/article/3088623

**Review by dingjiaweiww — APPROVED** [review]
LGTM
